vm/src/obj/objtype.rs

```diff
@@ -252,7 +261,7 @@ impl PyClass {
                 return Ok(());
             }
         }
-
+        zelf.slots.update_slot_func(attr_name.borrow_value());
```

Comment:
Looks like CPython checks that the name is dunderscored before calling update_slot, probably a micro-op but :shrug: https://github.com/python/cpython/blob/3.8/Objects/typeobject.c#L3324-L3326